File: sdk/python/openbrain_sdk/examples/mcp.py
from openbrain_sdk import OpenBrainMcpClient


def main() -> None:
    client = OpenBrainMcpClient.spawn_openbrain_mcp(openbrain_path="openbrain")
    try:
        pong = client.call_tool("openbrain.ping", {})
        print(pong["version"])
        pack = client.call_tool(
            "openbrain.memory.pack",
            {"scope": "project-alpha", "task_hint": "Summarize recent decisions"},
        )
        print(pack["pack"]["summary"])
    finally:
        client.close()


if __name__ == "__main__":
    main()

File: sdk/typescript/openbrain-sdk/examples/mcp.ts
import { OpenBrainMcpClient } from "../src/mcp.js";

async function main() {
  const client = await OpenBrainMcpClient.spawnOpenBrainMcp({
    openbrainPath: "openbrain",
  });

  try {
    await client.callTool("openbrain.ping", {});

    await client.callTool("openbrain.write", {
      objects: [
        {
          type: "artifact",
          id: "artifact-001",
          scope: "project-alpha",
          status: "canonical",
          spec_version: "0.1",
          tags: ["example", "mcp"],
          data: { summary: "Example artifact" },
          provenance: { actor: "mcp-example", type: "agent" },
        },
      ],
    });

    const pack = await client.callTool("openbrain.memory.pack", {
      scope: "project-alpha",
      task_hint: "Summarize relevant context",
    });
    console.log("pack summary:", pack.pack.summary);
  } finally {
    await client.close();
  }
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

